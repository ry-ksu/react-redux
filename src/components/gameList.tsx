export type GameProps = {
  key: number;
  img: string;
  title: string;
  creator: string;
  year: number;
  rating: number;
};

export const gameList: GameProps[] = [
  {
    key: 1,
    img: 'witcher3',
    title: 'The Witcher 3',
    creator: 'CD Projekt RED',
    year: 2015,
    rating: 8.6,
  },
  {
    key: 2,
    img: 'massEffect2',
    title: 'Mass Effect 2',
    creator: 'BioWare',
    year: 2010,
    rating: 9.3,
  },
  {
    key: 3,
    img: 'fahrenheit',
    title: 'Fahrenheit',
    creator: 'Quantic Dream',
    year: 2005,
    rating: 9.1,
  },
  {
    key: 4,
    img: 'rimWorld',
    title: 'RimWorld',
    creator: 'Ludeon Studios',
    year: 2013,
    rating: 9.3,
  },
  {
    key: 5,
    img: 'beholder',
    title: 'Beholder',
    creator: 'Warm Lamp Games',
    year: 2016,
    rating: 8.3,
  },
  {
    key: 6,
    img: 'nfs',
    title: 'NFS: Underground',
    creator: 'EA Games',
    year: 2003,
    rating: 8.7,
  },
  {
    key: 7,
    img: 'bannerSaga',
    title: 'The Banner Saga',
    creator: 'Stoic',
    year: 2013,
    rating: 9.1,
  },
  {
    key: 8,
    img: 'walkingDead',
    title: 'The Walking Dead',
    creator: 'Telltale Games',
    year: 2012,
    rating: 9.2,
  },
  {
    key: 9,
    img: 'heavyRain',
    title: 'Heavy Rain',
    creator: 'Quantic Dream',
    year: 2010,
    rating: 9.2,
  },
  {
    key: 10,
    img: 'warOfMine',
    title: 'This War of Mine',
    creator: '11 bit studios',
    year: 2014,
    rating: 8.7,
  },
];
