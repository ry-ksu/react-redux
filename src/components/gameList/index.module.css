.game-list-wrapper {
  background: #eceef2;
  min-height: calc(100vh - 242px);
}

.game-list__cards {
  max-width: 1200px;
  margin: auto;
  padding: 30px 5px;
  display: flex;
  justify-content: center;
  flex-wrap: wrap;
  gap: 15px;
}

.game-list__header {
  
}
