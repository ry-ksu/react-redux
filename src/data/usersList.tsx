export type UserProps = {
  name: string;
  birthday: string;
  eMail: string;
  enLvl: string;
  PDAgreement: boolean;
  sex: string;
  file: string;
};

export const userList: UserProps[] = [];
