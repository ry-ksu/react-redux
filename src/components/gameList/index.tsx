import React from 'react';

import { Loader } from '../loader';
import { useGlobalContext } from '../App';
import { GameWarning } from '../gameWarning';
import { GameList__header } from '../gameList__header/gameList__header';

import styles from './index.module.css';
import { GameCards } from 'components/gameCards';

type IGameListProps = {
  onClick: (e: React.MouseEvent<HTMLDivElement, MouseEvent>) => void;
};

export const GameList = (prop: IGameListProps) => {
  const { gamesState } = useGlobalContext();

  const firstSearch = 'Please enter a request...';
  const notData = 'No games have been created for this query yet. Try another request.';
  let content: JSX.Element | null;

  if (gamesState.isLoaded === 'not loaded') {
    content = GameWarning(firstSearch);
  } else if (gamesState.isLoaded === 'loaded' && gamesState.gamesCards.length === 0) {
    content = GameWarning(notData);
  } else if (gamesState.gamesCards.length != 0) {
    content = <GameCards onClick={prop.onClick} />;
  } else {
    content = null;
  }

  return (
    <div className={styles.cardsWrapper}>
      <div className={styles.gamesCards}>
        <GameList__header />
        {content}
      </div>
      <Loader isLoading={gamesState.isLoaded} />
    </div>
  );
};
