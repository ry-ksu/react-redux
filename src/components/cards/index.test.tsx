import React from 'react';
import { render, screen } from '@testing-library/react';
import { Cards } from '.';
import { gameList } from 'data/gameList';

describe('Cards component', () => {
  it('Cards renders with all data', () => {
    render(<Cards gameList={gameList} />);

    expect(screen.getByText(/The Witcher/i)).toBeInTheDocument();
    expect(screen.getByText(/mass effect 2/i)).toBeInTheDocument();
    expect(screen.getByText(/NFS:/i)).toBeInTheDocument();
    expect(screen.getByText(/this war of mine/i)).toBeInTheDocument();
  });

  it('Cards renders with one data', () => {
    const card = [
      {
        key: 10,
        img: 'warOfMine',
        title: 'This War of Mine',
        creator: '11 bit studios',
        year: 2014,
        rating: 8.7,
      },
    ];

    render(<Cards gameList={card} />);

    expect(screen.getByText(/This War of Mine/i)).toBeInTheDocument();
    expect(screen.getByText(/11 bit/i)).toBeInTheDocument();
    expect(screen.getByText(/2014/)).toBeInTheDocument();
    expect(screen.getByText('8.7')).toBeInTheDocument();
  });

  it('Cards renders without data', () => {
    render(<Cards gameList={[]} />);

    expect(screen.queryByText(/The Witcher/i)).toBeNull();
  });

  it('Cards snapshot', () => {
    const cards = render(<Cards gameList={gameList} />);

    expect(cards).toMatchSnapshot();
  });

  it('Cards empty snapshot', () => {
    const cards = render(<Cards gameList={[]} />);

    expect(cards).toMatchSnapshot();
  });
});
