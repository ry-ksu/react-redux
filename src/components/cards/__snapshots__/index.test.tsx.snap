// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Cards component Cards empty snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="cards"
      />
    </div>
  </body>,
  "container": <div>
    <div
      class="cards"
    />
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Cards component Cards snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="cards"
      >
        <div
          class="card"
        >
          <div
            class="card__img card__img_witcher3"
          />
          <div
            class="rating"
          >
            8.6
          </div>
          <div
            class="content"
          >
            <h3>
              The Witcher 3
            </h3>
            <h4>
              CD Projekt RED
            </h4>
            <h5>
              2015
            </h5>
          </div>
        </div>
        <div
          class="card"
        >
          <div
            class="card__img card__img_massEffect2"
          />
          <div
            class="rating"
          >
            9.3
          </div>
          <div
            class="content"
          >
            <h3>
              Mass Effect 2
            </h3>
            <h4>
              BioWare
            </h4>
            <h5>
              2010
            </h5>
          </div>
        </div>
        <div
          class="card"
        >
          <div
            class="card__img card__img_fahrenheit"
          />
          <div
            class="rating"
          >
            9.1
          </div>
          <div
            class="content"
          >
            <h3>
              Fahrenheit
            </h3>
            <h4>
              Quantic Dream
            </h4>
            <h5>
              2005
            </h5>
          </div>
        </div>
        <div
          class="card"
        >
          <div
            class="card__img card__img_rimWorld"
          />
          <div
            class="rating"
          >
            9.3
          </div>
          <div
            class="content"
          >
            <h3>
              RimWorld
            </h3>
            <h4>
              Ludeon Studios
            </h4>
            <h5>
              2013
            </h5>
          </div>
        </div>
        <div
          class="card"
        >
          <div
            class="card__img card__img_beholder"
          />
          <div
            class="rating"
          >
            8.3
          </div>
          <div
            class="content"
          >
            <h3>
              Beholder
            </h3>
            <h4>
              Warm Lamp Games
            </h4>
            <h5>
              2016
            </h5>
          </div>
        </div>
        <div
          class="card"
        >
          <div
            class="card__img card__img_nfs"
          />
          <div
            class="rating"
          >
            8.7
          </div>
          <div
            class="content"
          >
            <h3>
              NFS: Underground
            </h3>
            <h4>
              EA Games
            </h4>
            <h5>
              2003
            </h5>
          </div>
        </div>
        <div
          class="card"
        >
          <div
            class="card__img card__img_bannerSaga"
          />
          <div
            class="rating"
          >
            9.1
          </div>
          <div
            class="content"
          >
            <h3>
              The Banner Saga
            </h3>
            <h4>
              Stoic
            </h4>
            <h5>
              2013
            </h5>
          </div>
        </div>
        <div
          class="card"
        >
          <div
            class="card__img card__img_walkingDead"
          />
          <div
            class="rating"
          >
            9.2
          </div>
          <div
            class="content"
          >
            <h3>
              The Walking Dead
            </h3>
            <h4>
              Telltale Games
            </h4>
            <h5>
              2012
            </h5>
          </div>
        </div>
        <div
          class="card"
        >
          <div
            class="card__img card__img_heavyRain"
          />
          <div
            class="rating"
          >
            9.2
          </div>
          <div
            class="content"
          >
            <h3>
              Heavy Rain
            </h3>
            <h4>
              Quantic Dream
            </h4>
            <h5>
              2010
            </h5>
          </div>
        </div>
        <div
          class="card"
        >
          <div
            class="card__img card__img_warOfMine"
          />
          <div
            class="rating"
          >
            8.7
          </div>
          <div
            class="content"
          >
            <h3>
              This War of Mine
            </h3>
            <h4>
              11 bit studios
            </h4>
            <h5>
              2014
            </h5>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="cards"
    >
      <div
        class="card"
      >
        <div
          class="card__img card__img_witcher3"
        />
        <div
          class="rating"
        >
          8.6
        </div>
        <div
          class="content"
        >
          <h3>
            The Witcher 3
          </h3>
          <h4>
            CD Projekt RED
          </h4>
          <h5>
            2015
          </h5>
        </div>
      </div>
      <div
        class="card"
      >
        <div
          class="card__img card__img_massEffect2"
        />
        <div
          class="rating"
        >
          9.3
        </div>
        <div
          class="content"
        >
          <h3>
            Mass Effect 2
          </h3>
          <h4>
            BioWare
          </h4>
          <h5>
            2010
          </h5>
        </div>
      </div>
      <div
        class="card"
      >
        <div
          class="card__img card__img_fahrenheit"
        />
        <div
          class="rating"
        >
          9.1
        </div>
        <div
          class="content"
        >
          <h3>
            Fahrenheit
          </h3>
          <h4>
            Quantic Dream
          </h4>
          <h5>
            2005
          </h5>
        </div>
      </div>
      <div
        class="card"
      >
        <div
          class="card__img card__img_rimWorld"
        />
        <div
          class="rating"
        >
          9.3
        </div>
        <div
          class="content"
        >
          <h3>
            RimWorld
          </h3>
          <h4>
            Ludeon Studios
          </h4>
          <h5>
            2013
          </h5>
        </div>
      </div>
      <div
        class="card"
      >
        <div
          class="card__img card__img_beholder"
        />
        <div
          class="rating"
        >
          8.3
        </div>
        <div
          class="content"
        >
          <h3>
            Beholder
          </h3>
          <h4>
            Warm Lamp Games
          </h4>
          <h5>
            2016
          </h5>
        </div>
      </div>
      <div
        class="card"
      >
        <div
          class="card__img card__img_nfs"
        />
        <div
          class="rating"
        >
          8.7
        </div>
        <div
          class="content"
        >
          <h3>
            NFS: Underground
          </h3>
          <h4>
            EA Games
          </h4>
          <h5>
            2003
          </h5>
        </div>
      </div>
      <div
        class="card"
      >
        <div
          class="card__img card__img_bannerSaga"
        />
        <div
          class="rating"
        >
          9.1
        </div>
        <div
          class="content"
        >
          <h3>
            The Banner Saga
          </h3>
          <h4>
            Stoic
          </h4>
          <h5>
            2013
          </h5>
        </div>
      </div>
      <div
        class="card"
      >
        <div
          class="card__img card__img_walkingDead"
        />
        <div
          class="rating"
        >
          9.2
        </div>
        <div
          class="content"
        >
          <h3>
            The Walking Dead
          </h3>
          <h4>
            Telltale Games
          </h4>
          <h5>
            2012
          </h5>
        </div>
      </div>
      <div
        class="card"
      >
        <div
          class="card__img card__img_heavyRain"
        />
        <div
          class="rating"
        >
          9.2
        </div>
        <div
          class="content"
        >
          <h3>
            Heavy Rain
          </h3>
          <h4>
            Quantic Dream
          </h4>
          <h5>
            2010
          </h5>
        </div>
      </div>
      <div
        class="card"
      >
        <div
          class="card__img card__img_warOfMine"
        />
        <div
          class="rating"
        >
          8.7
        </div>
        <div
          class="content"
        >
          <h3>
            This War of Mine
          </h3>
          <h4>
            11 bit studios
          </h4>
          <h5>
            2014
          </h5>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
